code/src/main/java/com/onlinecontacttracing/servlets/get-data/GetNegativeUserInfoServlet.java

```diff
@@ -0,0 +1,62 @@
+package com.onlinecontacttracing.authentication;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import static com.googlecode.objectify.ObjectifyService.ofy;
+import com.onlinecontacttracing.storage.NegativeUser;
+import java.util.Optional;
+
+@WebServlet("/get-negative-user-info")
+public class GetNegativeUserInfoServlet extends CheckForApiAuthorizationServlet {
+
+  String getServletURIName() {
+    return "get-negative-user-info";
+  }
+  
+  void useCredential(Credential credential) {
+    Thread contactInfo = new Thread(new GetCalendarData(credential));
+
+    contactInfo.start();
+
+    try {
+      contactInfo.join();
+    } catch(Exception e) {
+      e.printStackTrace();
+    }
+
+    System.out.println("Negative User done getting info");
+  }
+
+  class GetCalendarData implements Runnable {
+    Credential credential;
+
+    public GetCalendarData(Credential credential) {
+      this.credential = credential;
+    }
+
+    public void run() {
+      // Get contacts from people api
+    }
+  }
```

Comment:
Let's move the actions into their own classes.

As a side note, I assume that there isn't output from these actions, right? They just store the output in datastore? If you do have to create an action that has a return type, I've got a few ideas.

Reply:
None of them return anything, however I pass an ArrayList to one of them that I later use outside of the runnable.